crates/ra_hir/src/nameres/collector.rs

```diff
@@ -675,6 +676,17 @@ fn resolve_submodule(
     }
 }
 
+fn normalize_attribute_path(file_path: &SmolStr) -> String {
```

Comment:
We could optimisitally try to avoid an allocation here, but, given this is already a cold path, this shouldn't matter.

Reply:
Fixed